Dockerfile-go-deps

```diff
@@ -5,7 +5,7 @@
 #
 # When this file is changed, run `bin/update-go-deps-shas`.
 
-FROM golang:1.14.2
+FROM golang:1.14.2-alpine
```

Comment:
Is this related to the change?

Reply:
Yeah the `actions/cache` space is very limited (5GB) so I'm reducing the base layer size by a few hundred megs doing this.

Reply:
Sounds good